plugin/pkg/scheduler/algorithm/predicates/error.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicates
+
+import "fmt"
+
+var (
+	ErrPodExceedsMaxPodNumber = newResourceUnfitError("PodExceedsMaxPodNumber")
+	ErrPodExceedsFreeCPU      = newResourceUnfitError("PodExceedsFreeCPU")
+	ErrPodExceedsFreeMemory   = newResourceUnfitError("PodExceedsFreeMemory")
```

Comment:
So if you want to just have static errors and not parameterize them at all (i.e., you don't want to put in requested/available etc)-- then you could just use errors.New() to create all of these.

Generally, if you make an error type then I expect it to be parameterizable and I expect that direct comparison is not the proper way to check it. (A type assertion + looking at its field(s) would be the proper way.)

I think given what you want to do, using errors.New() is less work.